pkg/resources/openstack/dns.go

```diff
@@ -67,6 +69,7 @@ func listDNSRecordsets(cloud openstack.OpenstackCloud, clusterName string) ([]*r
 			ID:   rr.ID,
 			Type: typeDNSRecord,
 			Deleter: func(cloud fi.Cloud, r *resources.Resource) error {
+				// TODO: not tested and this should have retry similar to what we have in another resources
 				return recordsets.Delete(cloud.(openstack.OpenstackCloud).DNSClient(), z.ID, rr.ID).ExtractErr()
```

Comment:
Fair, although I think we also have general retry logic in the code that calls Deleter.  So I _think_ we only need to retry if there's something special we can do.

Reply:
yeah actually I noticed that, it should not maybe have retry logic in Deleter. However, I am using some of these delete functions in openstacktasks - There the retry logic should be in place.